requirements.txt

```diff
@@ -1,2 +1,3 @@
 pytz>=2019.2
 numpy>=1.15.1
+Deprecated>=1.2.12
```

Comment:
Use `deprecated==1.2.0`. That worked for me. There must be some bug in the latest release.

Reply:
Maybe we should include a decorator of our own device as long as module `deprecated`.

Reply:
That's also an option. This is my first time working with the `deprecated` package, so I am not super familiar with the functionality.

Reply:
It seems like v1.2.0 behaves like we want it to though.

Reply:
Yeah, but that might lead to dependency clashes with downstream repositories.  Say, MNE wants to upgrade its version of module `deprecated` past v1.2.0.  This would clash with a strict requirement.

Reply:
MNE [copied](https://github.com/mne-tools/mne-python/blob/2d1e3dfe569c569bf0331b7faca41c483e8b8237/mne/utils/docs.py#L2716) deprecated from scikit-learn into its code base.  If module `deprecated` matures, they might switch thus leading to clashes with our fixed dependency.

Reply:
I think the behavior we find in module `deprecated` is actually correct.  The user of the application shouldn't see the warning, it's something that the developer should be aware of (see [here](https://www.lesinskis.com/python_deprecation_tutorial.html)).  So to check for the warnings one needs to run
```python
import warnings
warnings.simplefilter('default')
...
# raises warning
readerInstance.flavor
```

Reply:
Nice article. Yeah, that makes sense that the user wouldn't need or want to see the deprecation warning every time. I think as long as we make it clear in the `.flavor` docstring that it is deprecated, we will be good.